### ISSUE TITLE
Rework Notebook 7 docs accessibility section

### ISSUE THREAD

**Issue by andrii-i** [jupyter/notebook#6981]
Hey folks. Since the blog post was mentioned at the Jupyter accessibility call and feedback was requested, I thought I would share some thoughts.

Upon first read of the blog post, I did not notice anything accessibility-related but instead, the post links to the New features in Notebook v7 documentation which includes [a section on WCAG compliance][notebook-a11y-docs].
The section excerpts and links to [another blog post][Quantstack-blogpost] published a few months back.

## TL;DR

* I think it is great that folks are trying to make accessibility-related work and improvements more visible in the official documentation
* However, such documentation must be accurate, transparent, and aligned with the Jupyter community's inclusion goals. After rereading the blog post and based on all the accessibility work in flight and completed within Jupyter, I do not believe the current Notebook’s accessibility documentation meets those requirements (more details below, including suggestions on how to best present this information).

## The long version

Since the content in the Jupyter Notebook’s documentation is a small excerpt of the blog post in Medium - I will center my feedback on the latter as readers are directed to it for more information (from the Notebook docs).

1. The blog post includes biased and problematic language, such as:

    > 1. population has a disability that may impair their ability to use online services [^1]
    > 2. it requires accommodating a broad range of disabilities, including vision, motor, and cognitive impairments
    > 3. users who have specific needs

    The last phrase is a variation of the often-used "special needs" phrase. Us people with disabilities have the same access needs to physical and online spaces as everyone else. Therefore, they are not special needs but rather access needs.
    All these phrases somehow imply that something is wrong with us that needs fixing, while in reality, we cannot exercise our access rights (because it is by law a right) due to multiple systemic, attitudinal, and environmental barriers.

    > If we don’t want them to be excluded from learning sciences, technology, and engineering,

    In the same spirit, it deflects the responsibility from the Jupyter community to the disabled community. Our goal is disability justice and inclusion, and to do so, we must first acknowledge that the many barriers limiting the participation of people with disabilities in STEM are there because of us, the ones creating and maintaining the tools.
    We (the ones working on accessibility) are not the saviors of the disabled community nor preventing their exclusion, but rather are working towards removing existing barriers that have for a long time excluded people with disabilities from our communities.

    [^1]: Note that certain communities use the term impairment (for example, visually impaired), but this is highly contextualized, and we should always check how people prefer to define/refer to themselves.

2. Some information is not entirely accurate, such as:

    > Improving the accessibility of Jupyter had long been impeded by significant obstacles. The primary obstacle was that the text editor underlying the Jupyter Notebook (CodeMirror 5) had major accessibility issues.

     I would class the inaccessibility of CodeMirror as a considerable obstacle, but not the primary obstacle for Jupyter tools accessibility. Implying that the inaccessibility of CodeMirror was the primary obstacle for Jupyter accessibility is deflects responsibility from the Jupyter community and the many accessibility issues that have been reported and ignored for years.

     This can be supported through the multiple audits carried out on JupyterLab and Jupyter Notebook:

    * [Notebook v7.0.0 for WCAG 2.1 audit (March 2023)](https://github.com/jupyter/notebook/issues/6800)
    * [Notebook 7 prerelease keyboard navigation review (October 2022)](https://github.com/jupyter/notebook/issues/6595)
    * [JupyterLab Screen reader with keyboard shortcuts audit (June 2019)](https://github.com/jupyterlab/jupyterlab/issues/6573)
    * [JupyterLab v2.2.6 for WCAG 2.1 audit (November 2021)](https://github.com/jupyterlab/jupyterlab/issues/9399)
    * [JupyterLab 3.4.3 Zoom Audit (August 2022)](https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/34#issuecomment-1210168155)

    Through those audits, the community has unveiled many accessibility issues unrelated to CodeMirror, but to the JupyterLab and Jupyter Notebook UIs. Most of them result from considerable accessibility-related technical debt accumulated over the years, lack of accessibility knowledge, accessibility testing and UX/UI design (until recently), and minimal work with the disabled community to understand their needs and ensure we are building products that are usable and accessible to them.

3. Information is incomplete or not detailed enough, such as:

    > we made an automated audit of the accessibility of Notebook 7 with these changes, and found that the number of warnings and errors reported by Axe Auditor went down from several hundreds to a few dozen

    > we were able to make the Notebook 7 codebase pass the Axe Auditor tests with zero error or warning.

    While some automated audits, or rather automated diagnoses, were performed, there is little depth or information about them. Information that is missing is:

    * What was the methodology used to perform the audits?
    * There is no mention of [WCAG (Web Content Accessibility Guidelines)](https://www.w3.org/TR/WCAG21/) or which criteria were considered.
    * There is no mention that [automated tests only catch up to 30-50% of accessibility issues][limitations-testing].
    * How do these improvements align with WCAG conformance, and how do they impact people with disabilities?

    > It is estimated that 15% of the population has a disability

    This is a rather conservative number; some sources cite 15% others sources refer to numbers around 20-25% percent. For example, [the UK government cites 22% of the population has a disability][uk-disability-statistics], while the [CDC reports 27% for the US][us-disability-statistics], which can be significantly higher in emerging economies.

   If we account for people with temporary and situational disabilities, this number is also significantly higher.

4. Misleading information or data misrepresentation:

   The below image is presented on both the blog post and the Notebook documentation. I have already covered the lack of supporting information for such automated checks.
   There is insufficient context around this image (in text or alt-text); therefore, this is left up to the reader's interpretation.

   ![Axe Auditor results for Jupyter Notebook 7 from Quantstack blog post on Medium.  The results count shows a total of 0 issues with a diagnosis of best practices and some, non specified WCAG 2.1 criteria](https://miro.medium.com/v2/resize:fit:720/0*7imwXY7uST5QSNfi)

   Leaving the interpretation of this image up to the reader is problematic for a few reasons:

   1. For the non-accessibility savvy person, this image might be interpreted as the Jupyter Notebook is now fully accessible, which is not the case.
   2. This content is in an official Jupyter source, the Notebook documentation, and therefore, millions of users will take this at face value. This is problematic because it is inaccurate and could damage the community's trust in the project. I have seen other projects or tools making similar claims, only to be found that this was inaccurate, eventually leading to a loss of trust in the project and a break with the disabled community.
   3. There are a number of organizations with compliance requirements; having this type of information can lead to them making decisions that are not in the best interest of their users or the disabled community. And that can lead to legal issues for the organization and a loss of trust among its members.

    >  CodeMirror 6, a complete rewrite of the text editor with a strong focus on accessibility

    This is a claim I have tried to get more information on CodeMirror docs, discussion forums, and the such and have not found solid details on. I still have to find what that "strong focus of accessibility" means. Or when they say it is more accessible, I still do not know accessible how or to whom. I would like to know if these claims have been tested in Jupyter Notebook beyond the automated checks and, if so, how and what were the results.

    > The work by Johan Mabille at QuantStack on migrating JupyterLab to use CodeMirror 6 was funded by Two Sigma.

    AFAIK this was not a single individual work but a team effort. I would like to see the other contributors recognized for their work. I know of more folks involved in this effort, and only one individual is mentioned in the post; I assume this was an oversight?

## Recommendations 📝

Since it seems we all agree on making the accessibility work visible, we should do it in a way that transparently and accurately reports the accessibility and usability status of Jupyter Notebooks. I suggest doing a complete re-write of the [Jupyter Notebook v7 accessibility docs][notebook-a11y-docs] as a resource if its own rather than depending on an external blog post. By doing so, this could become the foundation of an evergreen resource accurately representing the status of the Jupyter Notebook’s accessibility and usability.
As for Notebook v7 there are other accessibility improvements that have been made and are not mentioned anywhere, the only one captured is the CodeMirror migration and I believe all the other contributions deserve mentioning too.

Upon rewriting, the documentation should contain at least the following information:

* A date/version (s) to which the report/data refers to
* WCAG conformance status
* Compatibility with browsers and assistive technologies
* Technical specifications
* Limitations
* Approaches to accessibility testing and auditing
* What we are working on in terms of accessibility

You can find an example of such an approach in the [JupyterLab Accessibility Statement](https://jupyter-accessibility.readthedocs.io/en/latest/resources/JupyterLab-a11y-statement.html). Which we hope to include in the JupyterLab documentation soon.

## Supporting resources 📖

* [CDC Disability impacts us all - statistics on disability in the US][us-disability-statistics]
* [UK disability statistics: Prevalence and life experiences][uk-disability-statistics]
* [Limitations of automated (accessibility) testing](https://design.sis.gov.uk/accessibility/testing/automated-testing-limitation)

[uk-disability-statistics]: https://commonslibrary.parliament.uk/research-briefings/cbp-9602/
[us-disability-statistics]: https://www.cdc.gov/ncbddd/disabilityandhealth/infographic-disability-impacts-all.html
[limitations-testing]: https://accessibility.blog.gov.uk/2017/02/24/what-we-found-when-we-tested-tools-on-the-worlds-least-accessible-webpage
[Quantstack-blogpost]: https://blog.jupyter.org/improving-the-accessibility-of-jupyter-6c695db518d3
[notebook-a11y-docs]: https://jupyter-notebook.readthedocs.io/en/latest/notebook_7_features.html#improved-web-content-accessibility-guidelines-wcag-compliance

_Originally posted by @trallard in https://github.com/jupyter/notebook-team-compass/issues/24#issuecomment-1635813804_

**Comment by srushtiipatill** [comment]
Hi @andrii-i 
I’m Srushti, a GSoC 2026 aspirant.  
I would love to work on improving this accessibility documentation.  
May I get assigned to this issue?

**Comment by andrii-i** [comment]
Hi @srushtiipatill. Nice to meet you, thank you for interest in contributing to Jupyter. 

We usually don't assign issues to people and don't require contributors to be assigned an issue before starting work on the PR addressing it so please feel free to just go ahead and open a PR addressing this issue or its' parts. 

(there are exceptions to this of course, for example we sometimes assign issues when we are preparing for a release and are looking to resolve blocking issues in a very limited amount of time)
